pynestml/codegeneration/builder.py

```diff
@@ -44,8 +57,29 @@ def __init__(self, target, options: Optional[Mapping[str, Any]]=None):
             self._target = ""
             raise InvalidTargetException()
 
+        self._redirect_output = copy.deepcopy(self.__class__._redirect_build_output)
         self._target = target
 
-    @abstractmethod
     def build(self) -> None:
+        if (self._redirect_output["default"]):
```

Comment:
Could this be moved to the `__init__` function instead? Then there is no need for the `main()` function.

Reply:
It should probably stay in ``build()`` so it redirects only stdout and stderr output that belongs strictly to the build stage.

``main()`` is very nondescriptive, how about ``build_()`` to indicate "private" method?